### PR TITLE
feat: add base SDK package

### PR DESCRIPTION
**Pull request by smartcontracts**
**Description**
Adds the base SDK package that I'll be using to implement all the stuff we've covered in the SDK project charter.

**Metadata**
- Fixes ENG-1744

### CODE REVIEW

File: packages/sdk/package.json [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    "sdk"
+  ],
+  "homepage": "https://github.com/ethereum-optimism/optimism-monorepo/tree/master/packages/sdk#readme",
+  "license": "MIT",
+  "author": "Optimism PBC",
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/ethereum-optimism/optimism-monorepo.git"
+  },
+  "devDependencies": {
```
Comment (tynes) [inline packages/sdk/package.json:31]: What is the reason behind using the `peerDependencies` for the ethers stuff?

**Review by tynes — APPROVED** [review]

**Comment by tynes** [comment]
This could also add the sdk package to changesets in the base `package.json` or could be done in a follow up PR

**Comment by tynes** [comment]
Safe to close https://github.com/ethereum-optimism/optimism/pull/1762?

File: packages/sdk/package.json (hunk 1 above)
Comment (smartcontracts) [reply packages/sdk/package.json:31]: Whoops, this was a holdover from a previous attempt at making the package.

**Comment by smartcontracts** [comment]
> Safe to close #1762?

Yep, closing.

**Pull request merged**